### PR TITLE
Finer OMP parallelisation

### PR DESCRIPTION
**Pull request by Luke-Pratley**
It is possible to not just use Openmp over rows of an image for the wavelet transform, but at the stage of applying filters. This PR proposes to do that.

### CODE REVIEW

File: cpp/sopt/wavelets/innards.impl.h [hunk 1]
```diff
@@ -116,13 +123,18 @@ void up_convolve_sum(Eigen::ArrayBase<T0> &result, Eigen::ArrayBase<T1> const &c
...
-    result(i + (is_even ? 0 : 1)) =
-        periodic_scalar_product(coeffs.head(Nlow), low_odd, i / 2 + odd_offset) +
-        periodic_scalar_product(coeffs.tail(Nhigh), high_odd, i / 2 + odd_offset);
+  auto const index_place_even = (is_even ? 1 : 0);
+  auto const index_place_odd = (is_even ? 0 : 1);
+#ifdef SOPT_OPENMP
+#pragma omp parallel for
+#endif
+  for (t_uint i = 0; i < result.size()/2; i++) {
+    result(2 *i + index_place_even) =
```
Comment (dpshelio) [inline cpp/sopt/wavelets/innards.impl.h]:
```suggestion
    result(2 * i + index_place_even) =
```
and in other places too...

Comment (Luke-Pratley) [reply cpp/sopt/wavelets/innards.impl.h]: I see, I will fix the lining soon

Comment (Luke-Pratley) [reply cpp/sopt/wavelets/innards.impl.h]: Actually, how did this pass the lint check?

Comment (Luke-Pratley) [reply cpp/sopt/wavelets/innards.impl.h]: Actually, how did this pass the lint check?

**Pull request merged**